src/controller/BrandController.ts

```diff
@@ -0,0 +1,62 @@
+import { getRepository } from 'typeorm';
+import { Body, Delete, Get, JsonController, OnUndefined, Param, Post, Patch } from 'routing-controllers';
+import { Brand } from '@entities/Brand';
+
+@JsonController()
+export class BrandController {
+
+  private brandRepository = getRepository(Brand);
+
+  @Get('/brands/')
+  async all() {
+    return await this.brandRepository.find();
+  }
+
+  @Get('/brands/:idOrName')
+    async one(@Param('idOrName') idOrName: string) {
+    let where : {name: string} | {id: number} = { name: idOrName };
+    if (idOrName.match(/[0-9]+/)) {
+      where = { id: parseFloat(idOrName) };
+    }
+    const brand = await this.brandRepository.findOne(where);
+    if (brand === undefined && !idOrName.match(/[0-9]+/)) {
```

Comment:
What's the point of the `.match` here  ?

Reply:
we have to scenario. we want to acquire a brand from the database but we weren't able to find one.
If we used id to search then we return undefined because we didn't found any brand corresponding to the id.
If we used a name (mostly used by the scrapping) and we don't have this brand in the database we create a new instance of this brand. The match is used to make a difference between search by id and search by name